src/Domain/DomainHelpers.hpp

```diff
@@ -102,17 +103,36 @@ template <size_t VolumeDim>
 class VolumeCornerIterator {
  public:
   VolumeCornerIterator() noexcept = default;
-  explicit VolumeCornerIterator(size_t index) noexcept : index_(index) {}
+  explicit VolumeCornerIterator(size_t local_index) noexcept
+      : local_index_(local_index) {}
+  explicit VolumeCornerIterator(
+      Index<VolumeDim> interior_multi_index,
```

Comment:
index_of_block  (and document that this is the same as the global_corner_index of the lower corner of the block)